src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.Unix.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System.Buffers;
+using System.IO;
+using System.Reflection;
+
+namespace System.Runtime.Loader
+{
+    public partial class AssemblyLoadContext
+    {
+        // Find satellite path using case insensitive culture name
+        internal static unsafe string? FindCaseInsensitiveSatellitePath(string parentDirectory, string cultureName, string assembly)
+        {
+            int bufferSize = Interop.Sys.GetReadDirRBufferSize();
+            byte[]? dirBuffer = null;
+            try
+            {
+                dirBuffer = ArrayPool<byte>.Shared.Rent(bufferSize);
+
+                fixed (byte* dirBufferPtr = dirBuffer)
+                {
+                    IntPtr dirHandle = Interop.Sys.OpenDir(parentDirectory);
+                    if (dirHandle != IntPtr.Zero)
+                    {
+                        try
+                        {
+                            Interop.Sys.DirectoryEntry dirent;
+                            while (Interop.Sys.ReadDirR(dirHandle, dirBufferPtr, bufferSize, out dirent) == 0)
+                            {
+                                if (dirent.InodeType != Interop.Sys.NodeType.DT_DIR)
+                                    continue;
+
+                                Span<char> nameBuffer = stackalloc char[Interop.Sys.DirectoryEntry.NameBufferSize];
+                                ReadOnlySpan<char> entryName = dirent.GetName(nameBuffer);
+
+                                if (cultureName.Length != entryName.Length)
+                                    continue;
+
+                                string entryNameString = entryName.ToString();
+
+                                if (!cultureName.Equals(entryNameString, StringComparison.InvariantCultureIgnoreCase))
+                                    continue;
+
+                                string assemblyPath = $"{parentDirectory}/{entryNameString}/{assembly}.dll";
+
+                                if (Internal.IO.File.InternalExists(assemblyPath))
+                                    return assemblyPath;
+                            }
+                        }
+                        finally
+                        {
+                            if (dirHandle != IntPtr.Zero)
```

Comment:
>if (dirHandle != IntPtr.Zero) [](start = 28, length = 29)

Is it possible dirHandle be equal IntPtr.Zero here?

Reply:
I had copied this code from `System.TimeZoneInfo.EnumerateFilesRecursively()`.  I didn't really look at changing it much because it was already tested and it worked.

Looks like you are right.